react-ui/src/deliveryontime/index.js

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from "react";
+import { BarChart, Bar, XAxis, YAxis, Legend, CartesianGrid, ResponsiveContainer, Tooltip, Rectangle } from 'recharts';
+import { Button, Form, Spinner } from "react-bootstrap";
+
+const DeliveryOnTimeDetail = ({ apiData }) => {
+    const [error, setError] = useState(false);
+    const [spinner, setSpinner] = useState(false);
+    const [sprintData, setSprintData] = useState([]);
+
+    useEffect(() => {
+        if (apiData && apiData.BV && apiData.story_points) {
+            const newData = apiData.BV.map((bvItem, index) => {
+                const spItem = apiData.story_points[index];
+                return {
+                    name: bvItem.milestoneName,
+                    bvCompleted: bvItem.bvCompleted,
+                    bvTotal: bvItem.bvTotal,
+                    spCompleted: spItem.spCompleted,
+                    totalPoints: spItem.totalPoints,
+                    remBV: bvItem.bvTotal - bvItem.bvCompleted,
+                    remSP: spItem.totalPoints - spItem.spCompleted
+                };
+            });
+            setSprintData(newData);
+        } else {
+            setError(true);
```

Comment:
I don't think we will ever reach this scenario. If we don't get the response, then it will fail in project/index.js